### PR TITLE
Metrics fixes and improvements

### PR DESCRIPTION
**Pull request by blazember**
- Add `tag(index)` and `tagValue(index)` to prevent allocation with
`readTags(BiConsumer)` in some cases
- Fix JmxPublisher to render discriminator tag correctly

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/internal/metrics/jmx/JmxPublisher.java [hunk 1]
```diff
@@ -191,16 +194,16 @@ static String escapeObjectNameValue(String name) {
...
                 module = moduleBuilder.toString();
             }
             String discriminator = descriptor.discriminator();
             String discriminatorValue = descriptor.discriminatorValue();
             if (discriminator != null && discriminatorValue != null) {
-                mBeanTags.append(escapeObjectNameValue(discriminator))
-                         .append('=')
-                         .append(escapeObjectNameValue(discriminatorValue));
+                int newTagIdx = tagCnt.getAndInc();
+                mBeanTags.append("tag").append(newTagIdx).append("=")
```
Comment (puzpuzpuz) [inline hazelcast/src/main/java/com/hazelcast/internal/metrics/jmx/JmxPublisher.java]: These changes should be probably covered by `JmxPublisherTest`. WDYT?

Comment (blazember) [reply hazelcast/src/main/java/com/hazelcast/internal/metrics/jmx/JmxPublisher.java]: @puzpuzpuz Added the coverage to the test

Comment (puzpuzpuz) [reply hazelcast/src/main/java/com/hazelcast/internal/metrics/jmx/JmxPublisher.java]: Thanks!

**Comment by blazember** [comment]
run-lab-run

**Pull request merged**